packages/browser-sdk/src/client.ts

```diff
@@ -293,29 +298,40 @@ export class BucketClient {
    *
    * @param options
    */
-  requestFeedback(options: Omit<RequestFeedbackOptions, "userId">) {
+  requestFeedback(options: RequestFeedbackData) {
     if (!this.context.user?.id) {
       this.logger.error(
-        "`requestFeedback` call ignored. No user context provided at initialization",
+        "`requestFeedback` call ignored. No `user` context provided at initialization",
+      );
+      return;
+    }
+
+    const featureId = "featureId" in options ? options.featureId : undefined;
+    const featureKey = "featureKey" in options ? options.featureKey : undefined;
+
+    if (!featureId && !featureKey) {
+      this.logger.error(
+        "`requestFeedback` call ignored. No `featureId` or `featureKey` provided",
       );
       return;
     }
 
     const feedbackData = {
-      featureId: options.featureId,
+      featureId,
+      featureKey,
       companyId:
         options.companyId ||
         (this.context.company?.id
           ? String(this.context.company?.id)
           : undefined),
       source: "widget" as const,
-    };
+    } as Feedback;
```

Comment:
Same issue with typescript and `union types`. I have to force it to `Feedback` to avoid it complaining about both id and key being in the object even if one is undefined and the other is not.